src/providers/on-chain-quote-provider.ts

```diff
@@ -367,7 +379,11 @@ export class OnChainQuoteProvider implements IOnChainQuoteProvider {
                 route,
                 functionName == 'quoteExactOutput' // For exactOut must be true to ensure the routes are reversed.
               )
-            : encodeMixedRouteToPath(route);
+            : encodeMixedRouteToPath(
+                route instanceof V2Route
+                  ? new MixedRouteSDK(route.pairs, route.input, route.output)
+                  : route
```

Comment:
Type wrangling that we have to do in order to support getting V2Routes on chain.

I'm not the biggest fan of this architecture...